### DISCUSSION TITLE
Standard Output and Error are mixed

### DISCUSSION THREAD

**Discussion by guidra-rev** [Tyrrrz/CliWrap#120] [Q&A]
I would like to redirect standard output and standard error to the console, in the same order as when I call the program manually in the console.

This example provided in the ReadMe.md, will mix output and error, so I get half a sentence from output, then half a sentence from error. I don't know who would ever need that.

```
await using var stdOut = Console.OpenStandardOutput();
await using var stdErr = Console.OpenStandardError();

var cmd = Cli.Wrap("foo") | (stdOut, stdErr);
await cmd.ExecuteAsync();
```

How can I get the same order as when I call manually ?

I tried calling synchronously instead, but still same issue

```
using var stdOut = Console.OpenStandardOutput();
using var stdErr = Console.OpenStandardError();

var cmd = Cli.Wrap("foo") | (stdOut, stdErr);
cmd.ExecuteAsync().GetAwaiter().GetResult();
```
As a workaround I use buffered execution, but this results in unnecessary copying, and I only see the output once it finished running
```
var result = Cli.Wrap("foo")
  .ExecuteBufferedAsync()
  .GetAwaiter().GetResult();

Console.Write(result.StandardOutput);
Console.Write(result.StandardError);
```

**Comment by Tyrrrz** [comment]
The example in the readme works with raw streams, so yeah race conditions are possible.
Try this instead:

```csharp
Cli.Wrap("foo") | (Console.WriteLine, Console.Error.WriteLine);
```

This will at least guarantee that the lines come in one piece

**Reply by guidra-rev** [reply]
Thank you, exactly what I needed
